src/main/java/org/elasticsearch/action/search/TransportSearchScrollAction.java

```diff
@@ -78,6 +84,34 @@ protected void doExecute(SearchScrollRequest request, ActionListener<SearchRespo
         }
     }
 
+    private void parseSource(SearchScrollRequest request) {
+        if (request.source() != null) {
```

Comment:
@martijnvg sorry, late reply. I push the change that I move parse methods to Helper class.
Does it make sense?